src/pyupgrade/plugins/redundant_open_modes.rs

```diff
@@ -0,0 +1,131 @@
+use rustpython_ast::{Constant, Expr, ExprKind, Located, Location};
+use rustpython_parser::lexer;
+use rustpython_parser::token::Tok;
+
+use crate::ast::helpers::{self, match_name_or_attr};
+use crate::ast::types::Range;
+use crate::autofix::Fix;
+use crate::check_ast::Checker;
+use crate::checks::{Check, CheckCode, CheckKind};
+use crate::source_code_locator::SourceCodeLocator;
+
+const OPEN_FUNC_NAME: &str = "open";
+
+enum OpenMode {
+    U,
+    Ur,
+    Ub,
+    RUb,
+    R,
+    Rt,
+    Wt,
+}
+
+impl OpenMode {
+    fn from_str(input: &str) -> Option<OpenMode> {
```

Comment:
Can you instead implement the `FromStr` trait here?

`DocstringConvention`, in our codebase, has an example of that:

```rust
impl FromStr for DocstringConvention {
    type Err = anyhow::Error;

    fn from_str(string: &str) -> Result<Self, Self::Err> {
        match string {
            "all" => Ok(DocstringConvention::All),
            "pep8" => Ok(DocstringConvention::PEP8),
            "numpy" => Ok(DocstringConvention::NumPy),
            "google" => Ok(DocstringConvention::Google),
            _ => Err(anyhow!("Unknown docstring convention: {}", string)),
        }
    }
}
```

Reply:
Ah my bad, of course, implemented `FromStr` 👍

Reply:
I do have one question regarding how I implemented the `FromStr` trait , would love to hear your take.
This is the `FromStr` trait implementation for `OpenMode`:
```rust
impl FromStr for OpenMode {
    type Err = anyhow::Error;

    fn from_str(string: &str) -> Result<Self, Self::Err> {
        match string {
            "U" => Ok(OpenMode::U),
            "Ur" => Ok(OpenMode::Ur),
            "Ub" => Ok(OpenMode::Ub),
            "rUb" => Ok(OpenMode::RUb),
            "r" => Ok(OpenMode::R),
            "rt" => Ok(OpenMode::Rt),
            "wt" => Ok(OpenMode::Wt),
            _ => Err(anyhow!("Unknown open mode: {}", string)),
        }
    }
}
```

So, when I match on this I have an empty match arm for the error case:

```rust
            match OpenMode::from_str(mode_param_value.as_str()) {
                Ok(mode) => {
                    checker.add_check(create_check(
                        expr,
                        mode_param,
                        mode.replacement_value(),
                        checker.locator,
                        checker.patch(&CheckCode::U015),
                    ));
                }
                Err(_) => {}
            }
```

Since I don't really need to do anything if I don't get an `Ok` from `from_str`, I just kept it as empty. But I didn't want to log with `eprintln!` though since I could have an open mode (like `"a"` for example) that doesn't need handling here.

I guess I'm wondering if an empty match arm for an error case is considered bad practice or not.

Reply:
Of course, `clippy` had the answers, used `if let` instead 👍